cmd/av/fetch.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"context"
+	"emperror.dev/errors"
+	"fmt"
+	"github.com/aviator-co/av/internal/config"
+	"github.com/aviator-co/av/internal/gh"
+	"github.com/aviator-co/av/internal/meta"
+	"github.com/fatih/color"
+	"github.com/shurcooL/githubv4"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+var fetchCmd = &cobra.Command{
+	Use:   "fetch",
+	Short: "fetch latest state from GitHub",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		repo, info, err := getRepoInfo()
+		if err != nil {
+			return err
+		}
+		branches, err := meta.ReadAllBranches(repo)
+		if err != nil {
+			return errors.Wrap(err, "failed to read av branch metadata")
+		}
+
+		client, err := gh.NewClient(config.GitHub.Token)
+		if err != nil {
+			return err
+		}
+
+		ctx := context.Background()
+		var cursor string
+		updatedCount := 0
+		for {
+			prsPage, err := client.RepoPullRequests(ctx, gh.RepoPullRequestOpts{
+				Owner:  info.Owner,
+				Repo:   info.Name,
+				After:  cursor,
+				States: []githubv4.PullRequestState{githubv4.PullRequestStateOpen},
+			})
+			if err != nil {
+				return errors.Wrap(err, "failed to fetch pull requests from GitHub")
+			}
+			if cursor == "" {
+				// only do this once at the start
+				_, _ = fmt.Fprint(
+					os.Stderr,
+					"Fetching ", color.CyanString(fmt.Sprintf("%d", prsPage.TotalCount)),
+					" open pull requests from GitHub...",
+					"\n",
+				)
+			}
+
+			for _, pr := range prsPage.PullRequests {
+				// TODO: maybe warn if local branch is not up-to-date with remote?
+				branchMeta, ok := branches[pr.HeadBranchName()]
+				if !ok {
+					logrus.WithField("branch", pr.HeadBranchName()).Debug("skipping PR for unknown local branch")
```

Comment:
suggest them that they should run git fetch?

Reply:
This is just so that we have all of the PR information for any branches you have locally that are being "tracked" by `av`. It's pretty likely that there will be PRs for branches that are not being locally tracked by `av` (e.g., because they are other people's PRs). Running `git fetch` wouldn't change that, because we still wouldn't have `av` metadata for the branch.